modules/gcp/cac-igm/cac-startup.sh.tmpl

```diff
@@ -38,156 +40,189 @@ get_credentials() {
+    log "--> Installing ldap_utils..."
+    RETRIES=5
+    while true; do
+        apt-get -qq update
+        apt-get -qq install ldap-utils
+        RC=$?
+        if [ $RC -eq 0 ] || [ $RETRIES -eq 0 ]; then
+            break
+        fi
+
+        log "--> ERROR: Failed to install ldap-utils. $RETRIES retries remaining..."
+        RETRIES=$((RETRIES-1))
+        sleep 5
+    done
+}
 
-    echo "Error installing ldap-utils. $RETRIES retries remaining..."
-    RETRIES=$((RETRIES-1))
-    sleep 5
-done
-
-echo '### Ensure AD account is available ###'
-TIMEOUT=1200
-until ldapwhoami \
-    -H ldap://${domain_controller_ip} \
-    -D ${ad_service_account_username}@${domain_name} \
-    -w $AD_SERVICE_ACCOUNT_PASSWORD \
-    -o nettimeout=1; do
-    if [ $TIMEOUT -le 0 ]; then
-        break
-    else
-        echo "Waiting for AD account ${ad_service_account_username}@${domain_name} to become available. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
-    fi
-    TIMEOUT=$((TIMEOUT-10))
-    sleep 10
-done
-
-# Check that the domain name can be resolved and that the LDAP port is accepting
-# connections. This could have been all done with the ldapwhoami command, but
-# due to a number of occasional cac-installation issues, such as "domain
-# controller unreachable" or "DNS error occurred" errors, check these explicitly
-# for logging and debug purposes.
-
-echo '### Ensure domain ${domain_name} can be resolved ###'
-TIMEOUT=1200
-until host ${domain_name}; do
-    if [ $TIMEOUT -le 0 ]; then
-        break
-    else
-        echo "Trying to resolve ${domain_name}. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
-    fi
-    TIMEOUT=$((TIMEOUT-10))
-    sleep 10
-done
-
-echo '### Ensure domain ${domain_name} port 636 is reacheable ###'
-TIMEOUT=1200
-until netcat -vz ${domain_name} 636; do
-    if [ $TIMEOUT -le 0 ]; then
-        break
+wait_for_dc() {
+    log "--> Ensure AD account is available..."
+    TIMEOUT=1200
+    until ldapwhoami \
+        -H ldap://${domain_controller_ip} \
+        -D ${ad_service_account_username}@${domain_name} \
+        -w $AD_SERVICE_ACCOUNT_PASSWORD \
+        -o nettimeout=1; do
+        if [ $TIMEOUT -le 0 ]; then
+            break
+        else
+            log "--> Waiting for AD account ${ad_service_account_username}@${domain_name} to become available. Retrying in 10 seconds... (Timeout in $TIMEOUT seconds)"
+        fi
+        TIMEOUT=$((TIMEOUT-10))
+        sleep 10
+    done
+
+    # Check that the domain name can be resolved and that the LDAP port is accepting
+    # connections. This could have been all done with the ldapwhoami command, but
+    # due to a number of occasional cac-installation issues, such as "domain
+    # controller unreachable" or "DNS error occurred" errors, check these explicitly
+    # for logging and debug purposes.
+    log "--> Ensure domain ${domain_name} can be resolved..."
+    TIMEOUT=1200
+    until host ${domain_name}; do
```

Comment:
just realized these 3 loops just continue if they timeout. we should write a log saying something like " time out but continue anyway" since there is no point just failing and stopping here.